client/wfnews-war/src/main/angular/src/app/components/draggable-panel/draggable-panel.component.ts

```diff
@@ -762,4 +827,40 @@ type = 'evac-order';
       return date.toLocaleDateString('en-US', options);
     } else return '';
   }
+
+  extractPolygonData(response) {
```

Comment:
An option here that may be more efficient is to store only the rings coordinates:

```
const points = []
for (const ring of response) {
   ... extract coords
}

then covert to a single polygon convex hull

`turf.convex(points);` or `window['turf'].convex(points);` if we dont have an import here.

You can return the generated convex hull polygon which will let you treat the zoom checks in a simple way